service/src/lib.rs

```diff
@@ -417,6 +439,10 @@ impl Service for SimpleService<rocket::Rocket<rocket::Build>> {
             .map_err(error::CustomError::new)?;
         Ok(())
     }
+
+    fn shutdown_handle(&mut self) -> Box<Option<tokio::sync::oneshot::Sender<bool>>> {
```

Comment:
Should we just make this `shutdown() -> Result<(), _>` (instead of returning a handle)? This may help each implementation to decide how to shut itself down.

Reply:
I also think that's better approach, and that's the first thing I tried, but couldn't figure this out: https://github.com/getsynth/shuttle/blob/c3610f60403506f760f0a2142a9d034c515cbebd/service/src/loader.rs#L60-L65 .
The issues is we're moving `service: Box<dyn Service>` into tokio thread, therefore we're no longer able to access it or its methods.